mule-maven-plugin/src/main/java/org/mule/tools/maven/mojo/CleanMojo.java

```diff
@@ -24,7 +28,25 @@
 public class CleanMojo extends AbstractMuleMojo {
 
   @Override
-  public void doExecute() {}
+  public void doExecute() {
+    Path targetDirPath = Paths.get(System.getProperty("java.io.tmpdir")).resolve("mmp-"
+        + project.getPluginArtifactMap().get("org.mule.tools.maven:mule-maven-plugin").getVersion() + "-ast-deps")
+        .resolve(CompileMojo.EXT_MODEL_LOADER_DEPENDENCIES_FOLDER);
```

Comment:
If it is no issue I would extract these lines into a method that could be shared with the creation of the temp dir.